### PR TITLE
Better host and port parsing. Bash in the shebang.

### PR DESCRIPTION
**Pull request by danielhoherd**
## Issue Description

Grafana failed to parse hostname from URI string where the password contained an `@` character.

The new syntax uses bash specific variable substitution. The running container is alpine with bash, so this is not a problem.

## Related Issue

Solves <https://github.com/astronomer/issues/issues/4452>

## Testing

```sh
#!/usr/bin/env bash


test_passwords=(
  postgres://postgres:postgres@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston
  postgres://user@:@:@:@:@:@user:password:@:@:@:@:@password@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston
)

parse_url() {
  echo "Testing url: $1"
  OLD_HOST="$(echo "$1" | awk -F@ '{print $2}' | awk -F: '{print $1}')"
  OLD_PORT="$(echo "$1" | awk -F@ '{print $2}' | awk -F: '{print $2}' | awk -F/ '{print $1}')"
  echo "OLD_HOST=${OLD_HOST}"
  echo "OLD_PORT=${OLD_PORT}"

  NEW_HOST="$(echo "${1##*@}" | awk -F: '{print $1}')"
  NEW_PORT="$(echo "${1##*:}" | awk -F/ '{print $1}')"
  echo "NEW_HOST=${NEW_HOST}"
  echo "NEW_PORT=${NEW_PORT}"
  echo
}

for url in "${test_passwords[@]}" ; do parse_url "$url" ; done ;
```

```
$ ./4452-pg-pw.sh
2022-04-14T07:19:00-0700
Testing url: postgres://postgres:postgres@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston
OLD_HOST=astronomer-postgresql.astronomer.svc.cluster.local
OLD_PORT=5432
NEW_HOST=astronomer-postgresql.astronomer.svc.cluster.local
NEW_PORT=5432

Testing url: postgres://user@:@:@:@:@:@user:password:@:@:@:@:@password@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston
OLD_HOST=
OLD_PORT=
NEW_HOST=astronomer-postgresql.astronomer.svc.cluster.local
NEW_PORT=5432
```

### CODE REVIEW

**Comment by pgvishnuram** [comment]
testing this now

**Review by rbankston — APPROVED** [review]
tested on an unbuntu 20 server with bash 5.0.17 and the string GF_DATABASE_URL=postgres://postgres:postgr@s@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston and it worked as expected. The previous version failed to produce anything

**Comment by pgvishnuram** [comment]
works as expected. we need to bump version.txt too

File: grafana/include/wait-for-db.sh [hunk 1]
```diff
@@ -11,8 +10,8 @@ wait_for_it() {
 }
 
 # Parse string like: postgres://postgres:postgres@astronomer-postgresql.astronomer.svc.cluster.local:5432/astronomer_houston
-POSTGRES_HOST=$(echo "${GF_DATABASE_URL}" | awk -F@ '{print $2}' | awk -F: '{print $1}')
-POSTGRES_PORT=$(echo "${GF_DATABASE_URL}" | awk -F@ '{print $2}' | awk -F: '{print $2}' | awk -F/ '{print $1}')
+POSTGRES_HOST="$(echo "${GF_DATABASE_URL##*@}" | awk -F: '{print $1}')"
```
Comment (andriisoldatenko) [inline grafana/include/wait-for-db.sh:13]: hm... we use similar code in houston-api https://github.com/astronomer/houston-api/blob/main/bin/entrypoint#L27-L28 FYI

Comment (danielhoherd) [reply grafana/include/wait-for-db.sh:13]: @andriisoldatenko you should update that houston code with these change.

Comment (danielhoherd) [reply grafana/include/wait-for-db.sh:13]: Or we could make something that is more reliable in python.

**Pull request merged**